src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -523,6 +531,11 @@ private AIState structureStep()
 
     private AIState decorationStep()
     {
+        if(job.getWorkOrder() == null)
```

Comment:
Doesn't seem like we need this, this check is done in checkIfExecute, also you didn't add it to any of the other steps.